dali/operators/sequence/optical_flow/turing_of/optical_flow_turing_test.cc

```diff
@@ -139,22 +139,20 @@ TEST_F(OpticalFlowTuringKernelTest, FlowVectorTest) {
           2.53125, 2.78125, 2.53125, 7.96875, 0.06250, 0.56250, 1.09375, 7.96875,
   };
   size_t width = 8, pitch = 10, height = 6;
-  float *input;
-  int16_t *tested;
-  CUDA_CALL(cudaMallocManaged(&input, test_data.size() * sizeof(float)));
-  CUDA_CALL(cudaMallocManaged(&tested, reference_data.size() * sizeof(int16_t)));
-  CUDA_CALL(
-          cudaMemcpy(input, test_data.data(), test_data.size() * sizeof(float), cudaMemcpyDefault));
-
-  optical_flow::kernel::EncodeFlowComponents(input, tested, pitch, width, height, 0);
+  DeviceBuffer<float> input;
+  DeviceBuffer<int16_t> tested;
+  std::vector<int16_t> tested_host;
+  input.from_host(test_data);
+  tested.resize(reference_data.size());
+  tested_host.resize(reference_data.size());
+
+  optical_flow::kernel::EncodeFlowComponents(input.data(), tested.data(), pitch, width, height, 0);
   CUDA_CALL(cudaDeviceSynchronize());
+  copyD2H(tested_host.data(), tested.data(), tested_host.size());
 
   for (size_t i = 0; i < reference_data.size(); i++) {
-    EXPECT_EQ(reference_data[i], tested[i]) << "Failed on index: " << i;
+    EXPECT_EQ(reference_data[i], tested_host[i]) << "Failed on index: " << i;
```

Comment:
This condition reads uninitialized memory (the padding between width and pitch) - either loop over width/height and check only the relevant values, or use cudaMemset to fill the array.

Reply:
Done